File: 4/Python/4.py
from Lenght import lenght

# Задача 1
print("Задача 1")
sentence = input("Введите предложение: ")
output = ''

for i in range(lenght(sentence) - 1):
    s1 = sentence[i]
    s2 = sentence[i + 1]

    if s1 == s2 and not output:
        output = f"{i + 1} {i + 2}"

if not output:
    print("Предложение не содержит повторяющихся символов")


# Задача 2
print("\nЗадача 2")
sentence = input("Введите предложение: ")

output = ''
i = 0
comma_counter = 0

while (i < lenght(sentence)) and comma_counter != 2:
    if sentence[i + 1] == ',':
        comma_counter += 1

    if (comma_counter == 1) and (sentence[i + 1] != ','):
        output += sentence[i + 1]

    i += 1

print(output)

# Задача 3
print("\nЗадача 3")
sentence = input("Введите предложение: ").lower()
i = 0

while sentence[i] not in "ст":
    i += 1
else:
    print(sentence[i] + " встретилось раньше")

File: 4/Pascal/4.pas
program f4;

uses Lenght;

var
  sentence, output: string;
  i, comma_counter: Integer;
  found, flag: Boolean;

begin
  // Задача 1
  WriteLn('Задача 1');
  Write('Введите предложение: ');
  ReadLn(sentence);
  found := False;

  for i := 1 to len(sentence) - 1 do
  begin
    if sentence[i] = sentence[i + 1] then
    begin
      WriteLn(i, ' ', i + 1);
      found := True;
      Break;
    end;
  end;

  if not found then
    WriteLn('Предложение не содержит повторяющихся символов');

  // Задача 2
  WriteLn;
  WriteLn('Задача 2');
  Write('Введите предложение: ');
  ReadLn(sentence);
  
  output := '';
  i := 0;
  flag := true;
  comma_counter := 0;
  
  while (i < len(sentence)) and flag do
  begin
    if sentence[i + 1] = ',' then
      comma_counter += 1;
  
    if (comma_counter = 1) and (sentence[i + 1] <> ',') then
      output += sentence[i + 1]
    else if comma_counter = 2 then
      flag := false;
  
    i += 1;
  end;
  
  WriteLn('Результат: ', output);

  // Задача 3
  WriteLn;
  WriteLn('Задача 3');
  Write('Введите предложение: ');
  ReadLn(sentence);
  sentence := LowerCase(sentence);
  found := False;
  i := 1;

  while (i <= Length(sentence)) and not found do
  begin
    if (sentence[i] = 'с') or (sentence[i] = 'т') then
    begin
      WriteLn(sentence[i], ' встретилось раньше');
      found := True;
    end
    else
      i += 1;
  end;
end.
